core/pagination/pagination.go

```diff
@@ -0,0 +1,38 @@
+package pagination
+
+import (
+	"math"
+)
+
+const (
+	DefaultPageSize = 50
+	DefaultPageNum  = 1
+)
+
+type Pagination struct {
+	PageNum    uint
```

Comment:
We could keep this to be in the same datatype as what we expect from proton. That way, we won't have to typecast it when using.

Reply:
Updated the datatype to be int32